plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/connection/Connection.java

```diff
@@ -11,41 +11,71 @@
 package org.jboss.tools.openshift.core.connection;
 
 import java.io.IOException;
-import java.io.InputStream;
-import java.net.HttpURLConnection;
 import java.net.MalformedURLException;
-import java.net.SocketTimeoutException;
 import java.net.URL;
 import java.util.List;
 
+import org.jboss.tools.common.databinding.ObservablePojo;
 import org.jboss.tools.openshift.common.core.connection.ConnectionType;
 import org.jboss.tools.openshift.common.core.connection.IConnection;
+import org.jboss.tools.openshift.core.auth.IAuthorizationClient;
 
 import com.openshift.client.IRefreshable;
 import com.openshift.client.OpenShiftException;
-import com.openshift.internal.client.utils.StreamUtils;
 import com.openshift3.client.IClient;
 import com.openshift3.client.ResourceKind;
+import com.openshift3.client.authorization.BearerTokenAuthorizationStrategy;
 import com.openshift3.client.model.IResource;
 import com.openshift3.internal.client.DefaultClient;
 
-public class Connection  implements IConnection, IRefreshable {
+public class Connection extends ObservablePojo implements IConnection, IRefreshable {
 
 	private final IClient client;
+	private IAuthorizationClient authorizer;
+	private String userName;
+	private String password;
+	private String token;
 	
-	public Connection(String url) throws MalformedURLException{
-		this(new DefaultClient(new URL(url)));
+	//TODO modify default client to take url and throw runtime exception
+	public Connection(String url, IAuthorizationClient authorizer) throws MalformedURLException{
+		this(new DefaultClient(new URL(url)), authorizer);
 	}
 	
-	public Connection(IClient client){
+	public Connection(IClient client, IAuthorizationClient authorizer){
 		this.client = client;
+		this.authorizer = authorizer;
 	}
-
-//	@Override
+	
+	public String getUsername(){
+		return this.userName;
+	}
+	
+	public void setUsername(String userName){
+		this.userName = userName;
+	}
+	public String getPassword(){
+		return this.password;
+	}
+	public void setPassword(String password){
+		this.password = password;
+	}
+	
+	@Override
 	public boolean connect() throws OpenShiftException {
-//		authorize();
-//		initializeCapabilities();
-		return true;
+		if(authorize()){
+//			initializeCapabilities();
+			return true;
+		} 
+		return false;
+	}
+
+	private boolean authorize() {
```

Comment:
@adietish See previous comment.  I think we should consider changing the method signature and maybe identifying the exceptions that it might throw.  These might be those already identified by v2